kernel/dev/serial/rs232.cpp

```diff
@@ -82,3 +90,14 @@ void px_rs232_init(uint16_t com_id) {
         "Panix Serial Output Debugger\n\n"
     );
 }
+
+px_ring_buff_t* px_rs232_init_buffer(int size) {
+    // Allocate space for the input buffer
+    read_buffer = (px_ring_buff_t*)malloc(sizeof(px_ring_buff_t));
+    // Initialize the ring buffer
+    if (px_ring_buffer_init(read_buffer, size) == 0) {
+        return read_buffer;
+    } else {
+        return NULL;
```

Comment:
No error message for failed initialization?

Reply:
The appropriate errno would be set by the ring buffer's init function so no need to duplicate work, right?

Reply:
Ah, right.